### PR TITLE
cockroach demo is secure by default

### LINKED ISSUE #7128: cli/demo: make `cockroach demo` use secure mode by default
PR: https://github.com/cockroachdb/cockroach/pull/46472

From release notes:
> The new `--unencrypted-localhost-http` flag for [`cockroach start`](../v20.1/cockroach-start.html) and [`cockroach start-single-node`](../v20.1/cockroach-start-single-node.html) forces the HTTP listener to bind to `localhost` addresses only and disables the TLS protocol. In secure clusters, this makes the Admin UI reachable with an `http://` URL without requiring certificate or CA setup. [#46472][#46472] {% comment %}doc{% endcomment %}

### PR DESCRIPTION
**Pull request by taroface**
Fixes #7128.

### CODE REVIEW

File: v21.1/movr-flask-setup.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ For debugging and development purposes, you can use the [`cockroach demo`](cockr
     ~~~ shell
     $ cockroach demo \
     --nodes=9 \
+    --insecure \
```
Comment (knz) [inline v21.1/movr-flask-setup.md:31]: Is this truly needed?

File: v20.1/cockroach-demo.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ The `cockroach demo` [command](cockroach-commands.html) starts a temporary, in-m
 
 The in-memory cluster persists only as long as the SQL shell is open. As soon as the shell is exited, the cluster and all its data are permanently destroyed. This command is therefore recommended only as an easy way to experiment with the CockroachDB SQL dialect.
 
-Each instance of `cockroach demo` loads a temporary [enterprise license](https://www.cockroachlabs.com/get-cockroachdb) that expires after an hour. To prevent the loading of a temporary license, set the `--disable-demo-license` flag.
+By default, `cockroach demo` starts in secure mode using TLS certificates to encrypt network communication. Each instance of `cockroach demo` loads a temporary [enterprise license](https://www.cockroachlabs.com/get-cockroachdb) that expires after an hour. To prevent the loading of a temporary license, set the `--disable-demo-license` flag.
```
Comment (knz) [inline v20.1/cockroach-demo.md:11]: Here and below: i think it's useful to mention that the demo web UI is not using TLS in any case.

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10955)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ericharmeling and @knz)

___
*[v20.1/cockroach-demo.md, line 11 at r1](https://reviewable.io/reviews/cockroachdb/docs/10955#-MfAQlEJQNZ4GzkBA6yV-r1-11:-MfExSy2ALa3MVAobPvq:bbx5n3v) ([raw file](https://github.com/cockroachdb/docs/blob/f77b436431aa7a5437271a30437a2644ef56ac21/v20.1/cockroach-demo.md#L11)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Here and below: i think it's useful to mention that the demo web UI is not using TLS in any case.
</blockquote></details>

Done (I think; let me know if I got this wrong).
___
*[v21.1/movr-flask-setup.md, line 31 at r1](https://reviewable.io/reviews/cockroachdb/docs/10955#-MfAQlEJQNZ4GzkBA6yY-r1-31:-MfEepky5k2BJYYIVNDP:b4a3z1h) ([raw file](https://github.com/cockroachdb/docs/blob/f77b436431aa7a5437271a30437a2644ef56ac21/v21.1/movr-flask-setup.md#L31)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

Is this truly needed?
</blockquote></details>

Hmm, this tutorial states that the demo cluster is insecure, so I assume the flag needs to make that true. @ericharmeling can you confirm that this change is appropriate for the tutorial?

File: v21.1/cockroach-demo.md [hunk 3]
```diff
@@ -178,10 +188,11 @@ When the SQL shell connects to the demo cluster at startup, it prints a welcome
...
 
 ~~~
-# Connection parameters:
-#   (webui)    http://127.0.0.1:8080/demologin?password=demo76950&username=demo
-#   (sql)      postgres://demo:demo76950@127.0.0.1:26257?sslmode=require
-#   (sql/unix) postgres://demo:demo76950@?host=%2Fvar%2Ffolders%2Fc8%2Fb_q93vjj0ybfz0fz0z8vy9zc0000gp%2FT%2Fdemo070856957&port=26257
+#   - Connection parameters:
+#     (webui)    http://127.0.0.1:8080/demologin?password=demo55826&username=demo
+#     (sql)      postgresql://demo:demo55826@127.0.0.1:26257/movr?sslmode=require
+#     (sql/jdbc) jdbc:postgresql://127.0.0.1:26257/movr?password=demo55826&sslmode=require&user=demo
```
Comment (ericharmeling) [inline v21.1/cockroach-demo.md:194]: > jdbc:postgresql://127.0.0.1:26257/movr?password=demo55826&sslmode=require&user=demo

Yo! First time seeing this! Love it.

File: v21.1/movr-flask-setup.md (hunk 1 above)
Comment (ericharmeling) [reply v21.1/movr-flask-setup.md:31]: I don't think we need this flag for the sample app.

Comment (ericharmeling) [reply v21.1/movr-flask-setup.md:31]: Below, we should instead update this:

```
    {% include copy-clipboard.html %}
    ~~~ shell
    $ cockroach sql --insecure --url='postgres://demo:<demo_password>@127.0.0.1:26257?sslmode=require' -f dbinit.sql
    ~~~

```

to this

```
    {% include copy-clipboard.html %}
    ~~~ shell
    $ cockroach sql --url='postgres://demo:<demo_password>@127.0.0.1:26257?sslmode=require' -f dbinit.sql
    ~~~
```

**Review by taroface — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10955)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @taroface)

___
*[v21.1/movr-flask-setup.md, line 31 at r1](https://reviewable.io/reviews/cockroachdb/docs/10955#-MfAQlEJQNZ4GzkBA6yY-r1-31:-MfF9G3d5zBeGuCM3Cw0:bnbj7t) ([raw file](https://github.com/cockroachdb/docs/blob/f77b436431aa7a5437271a30437a2644ef56ac21/v21.1/movr-flask-setup.md#L31)):*
<details><summary><i>Previously, ericharmeling (Eric Harmeling) wrote…</i></summary><blockquote>

Below, we should instead update this:

```
    {% include copy-clipboard.html %}
    ~~~ shell
    $ cockroach sql --insecure --url='postgres://demo:<demo_password>@127.0.0.1:26257?sslmode=require' -f dbinit.sql
    ~~~

```

to this

```
    {% include copy-clipboard.html %}
    ~~~ shell
    $ cockroach sql --url='postgres://demo:<demo_password>@127.0.0.1:26257?sslmode=require' -f dbinit.sql
    ~~~
```
</blockquote></details>

Done. Also changed the line "This command starts up an insecure, nine-node demo cluster." to "This command starts up a secure, nine-node demo cluster."

**Review by knz — APPROVED** [review]
> Reviewed 3 of 3 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10955)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @taroface)

**Review by ericharmeling — APPROVED** [review]
LGTM! Thanks Ryan!

**Pull request merged**